tests/test_std.rs

```diff
@@ -625,20 +680,44 @@ fn combinations_with_replacement() {
         ],
     );
     // Zero size
-    it::assert_equal(
-        (0..3).combinations_with_replacement(0),
-        vec![vec![]],
-    );
+    it::assert_equal((0..3).combinations_with_replacement(0), vec![vec![]]);
     // Zero size on empty pool
-    it::assert_equal(
-        (0..0).combinations_with_replacement(0),
-        vec![vec![]],
-    );
+    it::assert_equal((0..0).combinations_with_replacement(0), vec![vec![]]);
     // Empty pool
     it::assert_equal(
         (0..0).combinations_with_replacement(2),
         <Vec<Vec<_>>>::new(),
     );
+    it::assert_equal((1..3).unique_combinations(0), vec![vec![]]);
+}
+
+#[test]
+fn combinations_and_unique_combinations_has_all_unique_values() {
+    let mut rng = &mut rand::thread_rng();
+    let a = [1, 2, 3, 4, 5];
```

Comment:
Cool to have this kind of test. Just not sure if having non-determinism in tests is such a good idea, especially if the random test is only run once. (I may be wrong on this one though, if we already have precedence for this kind of test.)

Reply:
My biggest problem here was that i did not find out how you made sure quickcheck made vectors that was garanteed to contain duplicates.